src/core/public/notifications/toasts/error_toast.tsx

```diff
@@ -42,15 +43,37 @@ interface ErrorToastProps {
   i18nContext: () => I18nStart['Context'];
 }
 
+interface ErrorCause {
+  type: string;
+  reason: string;
+}
+
 interface RequestError extends Error {
-  body?: { attributes?: { error: { caused_by: { type: string; reason: string } } } };
+  body: {
+    attributes: {
+      error: Partial<ErrorCause & { caused_by: ErrorCause }>;
+    };
+  };
 }
 
-const isRequestError = (e: Error | RequestError): e is RequestError => {
-  if ('body' in e) {
-    return e.body?.attributes?.error?.caused_by !== undefined;
+const isRequestError = (e: unknown): e is RequestError =>
+  has(e, 'body.attributes.error.type') || has(e, 'body.attributes.error.caused_by');
+
+const getErrorText = (e: RequestError): string => {
+  const { error } = e.body.attributes;
+  let text = '';
+
+  if (error.type != null) {
+    text += `${error.type}\n`;
+    text += `${error.reason}\n`;
+  }
+
+  if (error.caused_by != null) {
+    text += `${error.caused_by.type}\n`;
+    text += `${error.caused_by.reason}\n`;
```

Comment:
I see https://github.com/elastic/kibana/pull/70404 and https://github.com/elastic/kibana/issues/81857, and I'm starting to feel like this tries to solve a need specific to a single app, as opposite that something that all the `toasts.addError` consumers may need. Am I assuming incorrectly here?

Reply:
👍 I see now that the fields I'm proposing were displayed in the `title` for #70404. We could do that, or us a custom component as you propose.

I'm happy to close this and implement an app-specific solution; if others need this functionality we can always reopen.

Reply:
Just to be clear: I'm fine with the implementation if this is something that is going to be widely used across multiple apps. I would just like to avoid app-specific code landing into core.

Reply:
@pgayvallet agreed! I'm happy to promote this to core when the need arises.